### PR TITLE
update openai smoke test

### PR DESCRIPTION
**Pull request by kazrael2119**
Fix a bug in extensible union as enum with nullable and update the spec for openai_modular smoke-test with latest rest api spec repo main branch.

### CODE REVIEW

File: packages/typespec-test/test/openai_modular/generated/typespec-ts/src/rest/outputModels.ts [hunk 1]
```diff
@@ -224,7 +258,7 @@ export interface ChoiceOutput {
   /** The log probabilities model for tokens associated with this completions choice. */
   logprobs: CompletionsLogProbabilityModelOutput | null;
   /** Reason for finishing */
-  finish_reason: CompletionsFinishReasonOutput | null;
+  finish_reason;
```
Comment (kazrael2119) [inline packages/typespec-test/test/openai_modular/generated/typespec-ts/src/rest/outputModels.ts:261]: image

File: packages/typespec-ts/src/utils/modelUtils.ts [hunk 2]
```diff
@@ -759,6 +767,9 @@ function getSchemaForModel(
   }
   for (const [propName, prop] of model.properties) {
     const restApiName = getWireName(dpgContext, prop);
+    if (restApiName === "finish_reason") {
+      prop;
+    }
```
Comment (qiaozha) [inline packages/typespec-ts/src/utils/modelUtils.ts:772]:
```suggestion
```

File: packages/typespec-ts/src/utils/modelUtils.ts [hunk 3]
```diff
@@ -388,37 +388,45 @@ function getSchemaForUnion(
...
+    });
+    values.push(variantType);
+    if (variantType.typeName) {
+      namedUnionMember = true;
+    }
   }
-  if (asEnum?.open) {
+  if (asEnum?.open && asEnum.members.size > 0) {
+    namedUnionMember = false;
     values = [];
```
Comment (MaryGao) [inline packages/typespec-ts/src/utils/modelUtils.ts:405]: just curious if we need to reset the values and namedUnionMember, why we go to foreach in line 392. Should we use open condition to separate them?

File: packages/typespec-ts/src/utils/modelUtils.ts [hunk 4]
```diff
@@ -388,37 +388,45 @@ function getSchemaForUnion(
...
+        namedUnionMember = true;
+      }
     }
   }
   const schema: any = {};
   if (values.length > 0) {
     schema.enum = values;
     const unionAlias =
-      asEnum?.open && asEnum?.kind
+      asEnum?.open && asEnum?.kind && !namedUnionMember
```
Comment (MaryGao) [inline packages/typespec-ts/src/utils/modelUtils.ts:421]: could you help me understand why the enum member having a name would impact the union alias?

File: packages/typespec-test/test/openai_modular/spec/examples/2024-05-01-preview/generated_audio_transcription_object.json [hunk 5]
```diff
@@ -0,0 +1,17 @@
+{
+  "operationId": "GetAudioTranscriptionAsResponseObject",
```
Comment (MaryGao) [inline packages/typespec-test/test/openai_modular/spec/examples/2024-05-01-preview/generated_audio_transcription_object.json:2]: i prefer not tracking these examples in smoke testing.

File: packages/typespec-ts/src/utils/modelUtils.ts (hunk 4 above)
Comment (qiaozha) [reply packages/typespec-ts/src/utils/modelUtils.ts:421]:
```
      union ExtensibleStringEnum {
        string;
        "a";
        "b";
      }
     model SimpleModel {
        propNullableRequired: ExtensibleStringEnum | null;
        propNullableFlattened: string | "a" | "b" | null;
     }
```
In the case, `getUnionAsEnum` for `propNullableRequired` and `propNullableFlattened` type will be a valid enum, which means `asEnum.open` is true and `asEnum.kind` has meaningful value. but we can't generate string type for `propNullableRequired` and `propNullableFlattened`. 
```
       export interface SimpleModel {
         propNullableRequired: ExtensibleStringEnum | null;
         propNullableFlattened: string | null;
       }
```
in the `propNullableRequired` case, we should get each value type, therefore, we need to see if the there's any named union in the member. 
in the `propNullableFlattened` case, we should leverage the `asEnum.kind` and `asEnum.nullable` to get the right typescript type.

File: packages/typespec-ts/src/utils/modelUtils.ts (hunk 3 above)
Comment (qiaozha) [reply packages/typespec-ts/src/utils/modelUtils.ts]: updated.

File: packages/typespec-test/test/openai_generic/generated/typespec-ts/src/rest/models.ts [hunk 6]
```diff
@@ -461,7 +461,7 @@ export interface CreateFineTuneRequest {
    * "davinci", or a fine-tuned model created after 2022-04-21 and before 2023-08-22. To learn more
    * about these models, see the [Models](/docs/models) documentation.
    */
-  model?: string;
+  model?: string | null;
```
Comment (qiaozha) [inline packages/typespec-test/test/openai_generic/generated/typespec-ts/src/rest/models.ts:464]: this is the typespec definition 
``model`?: string | "ada" | "babbage" | "curie" | "davinci" | null;`

File: packages/typespec-test/test/openai_non_branded/generated/typespec-ts/src/rest/models.ts [hunk 7]
```diff
@@ -460,7 +460,7 @@ export interface CreateFineTuneRequest {
    * "davinci", or a fine-tuned model created after 2022-04-21 and before 2023-08-22. To learn more
    * about these models, see the [Models](/docs/models) documentation.
    */
-  model?: string;
+  model?: string | null;
```
Comment (qiaozha) [inline packages/typespec-test/test/openai_non_branded/generated/typespec-ts/src/rest/models.ts:463]: same here, should be expected.

**Review by qiaozha — APPROVED** [review]

File: packages/typespec-ts/src/utils/modelUtils.ts (hunk 4 above)
Comment (MaryGao) [reply packages/typespec-ts/src/utils/modelUtils.ts:421]: maybe not a priority, but we may meet case for mixing both.

```
model SimpleModel {
        propNullableRequired: ExtensibleStringEnum | string | "a" | null;
}
```

Comment (qiaozha) [reply packages/typespec-ts/src/utils/modelUtils.ts:421]: `string | null` is our answer here.  but I think we should avoid writing typespec like that.

**Pull request merged**